### PR TITLE
(WIP) Documented all the encoder context options

### LINKED ISSUE #8505: [Serializer] Document the context options of every serializing format
We did this for XML in #7231, but we need to do the same for the other formats.

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #8505.

### CODE REVIEW

File: components/serializer.rst [hunk 1]
```diff
@@ -821,11 +899,72 @@ always as a collection.
...
 
+Context
+.......
+
+The ``encode()`` and ``encode()`` methods define an optional ``$context``
+argument to set the following configuration options:
+
+``xml_format_output``
+    **type**: ``boolean`` **default**: ``false``
+    If set to ``true`, formats the generated XML with line breaks and indentation.
```
Comment (wouterj) [inline components/serializer.rst]: typo, you forgot one backtick (resulting in error)

Comment (OskarStark) [inline components/serializer.rst]:
```suggestion
    If set to ``true``, formats the generated XML with line breaks and indentation.
```

File: components/serializer.rst [hunk 2]
```diff
@@ -775,16 +773,89 @@ always as a collection.
 
     Relying on the default value ``false`` is deprecated since Symfony 4.2.
 
+Context
+.......
+
+The ``encode()`` and ``encode()`` methods define an optional ``$context``
```
Comment (maxhelias) [inline components/serializer.rst]: You mean, encode and decode ?

**Comment by javiereguiluz** [comment]
I'm closing this because I've never used the Serializer, so I don't really know what I'm doing. Sorry.

**Pull request closed without merging**